deployer/src/deployment/mod.rs

```diff
@@ -79,13 +81,13 @@ impl Pipeline {
     /// executing/deploying built services. Two multi-producer, single consumer
     /// channels are also created which are for moving on-going service
     /// deployments between the aforementioned tasks.
-    fn new(kill_send: KillSender) -> Pipeline {
+    fn new(kill_send: KillSender, log_recorder: impl LogRecorder) -> Pipeline {
         let (queue_send, queue_recv) = mpsc::channel(QUEUE_BUFFER_SIZE);
         let (run_send, run_recv) = mpsc::channel(RUN_BUFFER_SIZE);
 
         let run_send_clone = run_send.clone();
 
-        tokio::spawn(async move { queue::task(queue_recv, run_send_clone).await });
+        tokio::spawn(async move { queue::task(queue_recv, run_send_clone, log_recorder).await });
```

Comment:
Nit: I know this is not from this PR, but I don't think the `async move` and the inner `.await` are required.